contracts/exchange/contracts/src/MixinMatchOrders.sol

```diff
@@ -262,6 +279,125 @@ contract MixinMatchOrders is
         return matchedFillResults;
     }
 
+    /// @dev Determines whether or not an order has been fully filled and returns the result.
+    /// @param order The order that should be checked.
+    /// @param fillResults The results of the most recent fill -- used to determine whether
+    ///                    or not the order was fully filled.
+    /// @return A boolean reflecting whether or not the order was fully filled
+    function _isFilled(
+        LibOrder.Order memory order,
+        LibFillResults.FillResults memory fillResults
+    )
+        internal
+        pure
+        returns (bool)
+    {
+        if (fillResults.takerAssetFilledAmount >= order.takerAssetAmount) {
```

Comment:
So this is the one mixin I haven't really touched, so maybe I'm wrong, but this doesn't feel right. I think `fillResults.takerAssetFilledAmount` is how much this order was filled in *this* fill, not total. I think you might actually need to keep a running tally of the fill amounts for left and right orders in your batch matching function.

Reply:
After some digging, I think that you're right about this one. I'll update!